# 2.0. Set up Librairies

In [1]:
import numpy as np
import pandas as pd
import os

# 2.1. Télécharger le fichier csv et l'ouvrir sur ton notebook

In [2]:
museums = pd.read_csv('/Users/juliensisavath2017/THP_loc/THP-2/S04/2/Project/1.Tour_de_France_des_musees/data.csv', sep = ";", encoding = 'latin-1')


In [3]:
museums

osm_id                                  name number  \
0     4540234591                               3 bis f    109   
1     4388745689                      A casa di Losari    NaN   
2     4426176194                      À l'homme volant     35   
3     7652598615           Abbadiale - Maison des arts    NaN   
4      121580330                                Abbaye    NaN   
...          ...                                   ...    ...   
3779  1348524503                    Vivarium du Moulin    NaN   
3780   577287841  Voyage au Temps des impressionnistes    NaN   
3781   145062184              Wines and Spirits Museum    NaN   
3782    82988725                              Xanxotea    NaN   
3783   277107550      Xylotheque - Kiosque Pedagogique    NaN   

                          street  postal_code              city country  \
0     Avenue du Petit Barthélémy      13617.0   Aix-en-Provence  France   
1                           T 30      20226.0         Belgodère  France   
2                     Grande Rue      72300.0  Sablé-sur-Sarthe  France   
3                 Rue du Gabizos      65400.0  Arras-en-Lavedan  France   
4                Rue de l'Abbaye      39210.0    Château-Chalon  France   
...                          ...          ...               ...     ...   
3779              Rue Principale      68610.0  Thann-Guebwiller  France   
3780                 Rue de Léry      95430.0   Auvers-sur-Oise  France   
3781          Corniche Bonaparte      83150.0            Bandol  France   
3782                    La Place      64240.0          Isturits  France   
3783     Impasse de la Haie Vive      45590.0  Saint-Cyr-en-Val  France   

     country_code        lat       lon                    website  \
0              fr  43.521207  5.439875       http://www.3bisf.com   
1              fr  42.638732  9.007660                        NaN   
2              fr  47.839656 -0.334043                        NaN   
3              fr  42.990826 -0.128057  https://www.abbadiale.fr/   
4              fr  46.753866  5.624296                        NaN   
...           ...        ...       ...                        ...   
3779           fr  47.942258  7.149613                        NaN   
3780           fr  49.072057  2.166167                        NaN   
3781           fr  43.128031  5.749484                        NaN   
3782           fr  43.364834 -1.203580   http://www.xanxotea.com/   
3783           fr  47.825888  1.967795                        NaN   

                  email         phone  fax  \
0     contact@3bisf.com  3.344216e+10  NaN   
1                   NaN           NaN  NaN   
2                   NaN  3.324395e+10  NaN   
3                   NaN  3.363317e+10  NaN   
4                   NaN           NaN  NaN   
...                 ...           ...  ...   
3779                NaN           NaN  NaN   
3780                NaN           NaN  NaN   
3781                NaN           NaN  NaN   
3782                NaN           NaN  NaN   
3783                NaN           NaN  NaN   

                                         tags  \
0             osm:museum;art:art contemporain   
1                                  osm:museum   
2     osm:museum;type:centre d'interpretation   
3                                  osm:museum   
4                 osm:museum;art:architecture   
...                                       ...   
3779                               osm:museum   
3780                               osm:museum   
3781                               osm:museum   
3782                               osm:museum   
3783                               osm:museum   

                                            description        date_added  \
0                                                   NaN  44116,0076388889   
1                                                   NaN  43338,9393981481   
2                                                   NaN  42644,4636574074   
3                                                

In [4]:
museums.head()

osm_id                         name number                      street  \
0  4540234591                      3 bis f    109  Avenue du Petit Barthélémy   
1  4388745689             A casa di Losari    NaN                        T 30   
2  4426176194             À l'homme volant     35                  Grande Rue   
3  7652598615  Abbadiale - Maison des arts    NaN              Rue du Gabizos   
4   121580330                       Abbaye    NaN             Rue de l'Abbaye   

   postal_code              city country country_code        lat       lon  \
0      13617.0   Aix-en-Provence  France           fr  43.521207  5.439875   
1      20226.0         Belgodère  France           fr  42.638732  9.007660   
2      72300.0  Sablé-sur-Sarthe  France           fr  47.839656 -0.334043   
3      65400.0  Arras-en-Lavedan  France           fr  42.990826 -0.128057   
4      39210.0    Château-Chalon  France           fr  46.753866  5.624296   

                     website              email         phone  fax  \
0       http://www.3bisf.com  contact@3bisf.com  3.344216e+10  NaN   
1                        NaN                NaN           NaN  NaN   
2                        NaN                NaN  3.324395e+10  NaN   
3  https://www.abbadiale.fr/                NaN  3.363317e+10  NaN   
4                        NaN                NaN           NaN  NaN   

                                      tags description        date_added  \
0          osm:museum;art:art contemporain         NaN  44116,0076388889   
1                               osm:museum         NaN  43338,9393981481   
2  osm:museum;type:centre d'interpretation         NaN  42644,4636574074   
3                               osm:museum         NaN  44120,0466782407   
4              osm:museum;art:architecture         NaN  43102,5491898148   

  wikidata  
0      NaN  
1      NaN  
2      NaN  
3      NaN  
4      NaN

# 2.2. Répondre aux questions posées par ton manager

## 2.2.1. Combien y-a-t-il de musées en France métropolitaine ?

In [5]:
len(museums)

3784

In [6]:
nombre_total_musee = museums['osm_id'].value_counts(dropna=False).size
print(nombre_total_musee)

3784


### Réponse : Selon deux méthodes, il y en a 3'784.

## 2.2.2. Dans quelle(s) ville(s) y-a-t-il de plus de musées ?

In [7]:
museums.describe(include="all")

osm_id                  name number      street    postal_code  \
count   3.784000e+03                  3784    743        3771    3728.000000   
unique           NaN                  3466    197        2932            NaN   
top              NaN  Musée des Beaux-Arts      1  Grande Rue            NaN   
freq             NaN                    23     73          34            NaN   
mean    2.193306e+09                   NaN    NaN         NaN   48499.273069   
std     2.521168e+09                   NaN    NaN         NaN   27982.341434   
min     5.107373e+06                   NaN    NaN         NaN    1000.000000   
25%     1.451896e+08                   NaN    NaN         NaN   25582.500000   
50%     8.226927e+08                   NaN    NaN         NaN   49730.000000   
75%     3.982632e+09                   NaN    NaN         NaN   71570.000000   
max     8.394269e+09                   NaN    NaN         NaN  590007.000000   

         city country country_code          lat          lon  \
count    3782    3784         3784  3784.000000  3784.000000   
unique   2295       2            2          NaN          NaN   
top     Paris  France           fr          NaN          NaN   
freq      175    3776         3776          NaN          NaN   
mean      NaN     NaN          NaN    46.531075     2.724237   
std       NaN     NaN          NaN     2.234572     2.897485   
min       NaN     NaN          NaN    41.498794    -5.129406   
25%       NaN     NaN          NaN    44.542494     0.570318   
50%       NaN     NaN          NaN    46.513252     2.563655   
75%       NaN     NaN          NaN    48.639156     4.984839   
max       NaN     NaN          NaN    51.053756     9.488302   

                            website                   email         phone  \
count                          1636                     301  1.211000e+03   
unique                         1585                     295           NaN   
top     http://www.ville-orleans.fr  musees@ville.angers.fr           NaN   
freq                              4                       3           NaN   
mean                            NaN                     NaN  3.637847e+10   
std                             NaN                     NaN  9.501361e+10   
min                             NaN                     NaN  8.250029e+08   
25%                             NaN                     NaN  3.325117e+10   
50%                             NaN                     NaN  3.338552e+10   
75%                             NaN                     NaN  3.349082e+10   
max                             NaN                     NaN  3.320256e+12   

        fax        tags description        date_added  wikidata  
count   0.0        3782         226              3775      1147  
unique  NaN         119         206              3331      1133  
top     NaN  osm:museum    Ecomusée  43940,5848958333  Q3330384  
freq    NaN        2175          19                76         2  
mean    NaN         NaN         NaN               NaN       NaN  
std     NaN         NaN         NaN               NaN       NaN  
min     NaN         NaN         NaN               NaN       NaN  
25%     NaN         NaN         NaN               NaN       NaN  
50%     NaN         NaN         NaN               NaN       NaN  
75%     NaN         NaN         NaN               NaN       NaN  
max     NaN         NaN         NaN               NaN       NaN

In [8]:
museums["city"].value_counts()

Paris              175
Lyon                33
Marseille           28
Grenoble            22
Toulouse            20
                  ... 
Felleries            1
Pays-de-Clerval      1
Ploubazlanec         1
Soueix               1
Sarrebourg           1
Name: city, Length: 2295, dtype: int64

In [9]:
museums['city'].value_counts().to_dict()

{'Paris': 175,
 'Lyon': 33,
 'Marseille': 28,
 'Grenoble': 22,
 'Toulouse': 20,
 'Bordeaux': 20,
 'Nice': 16,
 'Strasbourg': 16,
 'Rouen': 15,
 'Lille': 15,
 'Aix-en-Provence': 13,
 'Montpellier': 13,
 'Reims': 13,
 'Angers': 13,
 'Dijon': 11,
 'Avignon': 11,
 'Colmar': 11,
 'Besançon': 10,
 'Perpignan': 10,
 'Tours': 10,
 'Digne-les-Bains': 9,
 'Saint-Étienne': 9,
 'Rochefort': 9,
 'Auxerre': 9,
 'Orléans': 9,
 'Nantes': 9,
 'Nîmes': 9,
 'Mulhouse': 8,
 'Rodez': 8,
 'Monaco': 8,
 'La Rochelle': 8,
 'Toulon': 8,
 'Commana': 8,
 'Cherbourg-en-Cotentin': 8,
 'Thiers': 8,
 'Limoges': 8,
 'Tourcoing': 7,
 'Chambéry': 7,
 'Cholet': 7,
 'Clermont-Ferrand': 7,
 'Compiègne': 7,
 'Briançon': 7,
 'Chamonix-Mont-Blanc': 7,
 'Nancy': 7,
 'Versailles': 7,
 'Blois': 7,
 'Mougins': 6,
 'Antibes': 6,
 'Chantilly': 6,
 'Draguignan': 6,
 'Caen': 6,
 'Laval': 6,
 'Sarlat-la-Canéda': 6,
 'Le Mans': 6,
 'Bourges': 6,
 'Albi': 6,
 'Pernes-les-Fontaines': 6,
 "Villeneuve-d'Ascq": 6,
 'Lourdes': 6,
 'Pau': 6,

In [10]:
museums['city'].value_counts().idxmax()

'Paris'

### Réponse : Selon quatre méthodes, Paris possède le plus de musées.


## 2.2.3. Quel est le nombre moyen de musées par ville ?

In [11]:
museums["city"].value_counts().mean()

1.647930283224401

## 2.2.4. Quel est le nombre médian de musées par ville ?

In [12]:
museums["city"].value_counts().median()

1.0

## 2.2.5. Comment sont répartis les musées par type (en pourcentage) ?

In [13]:
museums['tags'].value_counts(normalize = 'true')*100

osm:museum                                                                               57.509254
osm:museum;type:ecomusee                                                                 10.232681
osm:museum;type:musee technique et industriel                                             5.129561
osm:museum;type:musee d'histoire                                                          3.490217
osm:museum;art:architecture                                                               3.199365
                                                                                           ...    
osm:museum;art:art medieval                                                               0.026441
osm:museum;aart:architecture                                                              0.026441
osm:museum;gender:musee de couple;type:maison d'artiste                                   0.026441
osm:museum;art:art moderne;art:art contemporain;art:art brut;type:musee de beaux-arts     0.026441
osm:museum

In [14]:
pourcentage100 = museums["tags"].value_counts(normalize=True).mul(100).round(2).astype(str)+'%'
pourcentage100

osm:museum                                                                               57.51%
osm:museum;type:ecomusee                                                                 10.23%
osm:museum;type:musee technique et industriel                                             5.13%
osm:museum;type:musee d'histoire                                                          3.49%
osm:museum;art:architecture                                                                3.2%
                                                                                          ...  
osm:museum;art:art medieval                                                               0.03%
osm:museum;aart:architecture                                                              0.03%
osm:museum;gender:musee de couple;type:maison d'artiste                                   0.03%
osm:museum;art:art moderne;art:art contemporain;art:art brut;type:musee de beaux-arts     0.03%
osm:museum;art:architecture;type:musee t

## 2.2.6. Combien y-a-t-il de musées dont le nom commence par "Château" ?

In [15]:
museums['name'].str.startswith('Château').value_counts()

False    3727
True       57
Name: name, dtype: int64

In [16]:
museums[museums['name'].str.startswith('Château')]

osm_id                                               name number  \
270  1684479871                                            Château    NaN   
271   192623749                                    Château d'Arsac    NaN   
272    70789657                                 Château d'Asnières     89   
273   619219383                              Château de Bernicourt    NaN   
274    70279441                               Château de Bucheneck    NaN   
275    80559366                                      Château de By     12   
276  4478123325                           Château de Châteaubriant    NaN   
277   396793983                               Château de Désaignes    NaN   
278    95903744                                   Château de Dinan    NaN   
279   303480678                                   Château de Fitou    NaN   
280  3241536133                                    Château de Foix    NaN   
281   369893639                                 Château de Gramont    NaN   
282   125735941                                    Château de Joux    NaN   
283  2330409966                         Château de la Bâtie d'Urfé    NaN   
284    80773694                             Château de la Terrasse    NaN   
285  2018713809  Château de la Verrerie - Musée de l'Homme et d...    NaN   
286   420294471                                  Château de Lacaze    NaN   
287   145189098                               Château de l'Hermine    NaN   
288  2433718893                               Château de Malbrouck    NaN   
289    68566900                             Château De Monbazillac    NaN   
290   128925665                            Château de Monte-Cristo    NaN   
291    31101634                                 Château de Quintin    NaN   
292   146142380                             Château de Rambouillet    NaN   
293   120856493         Château de Réghat, Musée de Maisons-Alfort     34   
294  3594399915                           Château de Roquetaillade    NaN   
295    80773777                                Château de Rubelles    NaN   
296    68962958                     Château de Saché, Musée Balzac    NaN   
297    73850178                  Château de Saint-Amand-en-Puisaye    NaN   
298   116073486                      Château de Savigny lès Beaune    NaN   
299   391143508                                Château de Sedières    NaN   
300   528996147                                Château de Suscinio    NaN   
301   188151245                                  Château de Termes      2   
302   129454734                                 Château de Thierry    NaN   
303  4265122872                              Château de Versailles    NaN   
304   129667034                              Château de Villeneuve    NaN   
305    61565556                      Château d'eau Charles Laganne    NaN   
306  6468984786                       Château d'Égreville massenet    NaN   
307   113069661                             Château des Bouillants    NaN   
308   128925438                                       Château d'If    NaN   
309   102960021                                    Château d'Oiron    NaN   
310    62782286                               Château du Clos-Lucé      2   
311  7944701712            Château du Patys - Maison d'Hervé Bazin    NaN   
312   251862484                                Château du Sailhant    NaN   
313    63886282                                   Château Grimaldi    NaN   
314    63820570                                     Château Lafont     18   
315  6528325404                 Château Musée des Comtes du Perche    NaN   
316  3594413678     Château Pichon Longueville Comtesse de Lalande    NaN   
317    41159704                                      Château Vodou    NaN   
318  2383345899                           Château-Abbaye de Cassan    NaN   
319   145189774                                   Château-Gaillard      2   
320   128860574                                     Château-Mairie    NaN   
321  

In [17]:
len(museums[museums['name'].str.startswith('Château')])

57

In [18]:
len(museums.loc[museums['name'].str.startswith('Château')])

57

### Réponse : Selon quatre méthodes, il y a 57 musées dont le nom commencd par "Chateau".

## 2.2.7. Pour combien de musées dispose-t-on de l'adresse du site web ?

In [19]:
museums['website'].value_counts()

http://www.ville-orleans.fr                                                                          4
https://troismuseesprissac.jimdo.com/                                                                3
http://musees.agglo-annecy.fr                                                                        3
http://www.musees-langres.fr                                                                         3
http://www.bourges.fr/culture/html/musexpo.htm                                                       3
                                                                                                    ..
https://www.moulin-vallisclausa.com/                                                                 1
http://www.bourges.fr/site/musee-berry                                                               1
https://www.lyon-france.com/Je-decouvre-Lyon/culture-et-musees/Musees/atelier-de-tissage-mattelon    1
http://culture.marseille.fr/node/613                                     

In [20]:
museums['website'].value_counts().sum()

1636

In [21]:
len(museums["website"].dropna())

1636

In [22]:
len(museums.website.dropna())

1636

In [23]:
museums["website"].count()

1636

In [24]:
len(museums['website']) - museums['website'].isnull().sum()

1636

### Réponse : Selon six méthodes, 1636 musées possèdent un email

## 2.2.8. Quel département français possède le plus de musées sur son territoire ?

In [25]:
museums['postal_code'].value_counts()

75005.0    24
38000.0    21
75007.0    16
33000.0    16
75004.0    15
           ..
12420.0     1
89100.0     1
33990.0     1
55101.0     1
80260.0     1
Name: postal_code, Length: 2023, dtype: int64

In [26]:
museums["departement"] = museums['postal_code'].astype(str).str[0:2]
museums["departement"].value_counts()

75    180
59     97
33     94
71     91
13     90
     ... 
93     14
23     11
90     10
48      9
98      8
Name: departement, Length: 88, dtype: int64

In [27]:
departement = museums['postal_code'].dropna()
departement = departement.astype(int).astype(str).str.slice(stop=2)
departement

0       13
1       20
2       72
3       65
4       39
        ..
3779    68
3780    95
3781    83
3782    64
3783    45
Name: postal_code, Length: 3728, dtype: object

In [28]:
departement = museums['postal_code'].dropna()
departement = departement.astype(int).astype(str).str.slice(stop=2)
departement.value_counts().idxmax()

'75'

### Réponse : Selon six méthodes, l'île de France

## 2.2.9. Quel département français possède le moins de musées sur son territoire ?

In [29]:
departement = museums['postal_code'].dropna()
departement = departement.astype(int).astype(str).str.slice(stop=2)
departement.value_counts().idxmin()

'98'

### Réponse : Les départements d'outre mer, 987 : Polynésie française. 988 : Nouvelle-Calédonie. 989 : Île de Clipperton.

## 2.2.10. Combien de musées ont "Napoléon" dans leur nom ?

In [30]:
museums["name"].str.contains("Napoléon").value_counts()

False    3780
True        4
Name: name, dtype: int64

# 2.3. Proposer deux axes d'analyse supplémentaires

## Un site qui donne la fréquentation de tous les musées français

[Fréquentation des musées de France](https://data.culture.gouv.fr/explore/dataset/frequentation-des-musees-de-france/information/)


## Qui visite les musées

[Qui visite les musées](https://doc.ocim.fr/LO/LO035/LO.35%284%29-pp.16-23.pdf)